apps/client/sttr_build/transform.ts

```diff
@@ -0,0 +1,82 @@
+import { readJson, writeJson } from "https://deno.land/std/fs/mod.ts";
+import { join } from "https://deno.land/std/path/mod.ts";
+
+import sttrbuild from "./parser.js";
+import { APIConfig } from "./types.d.ts";
+
+// XXX npm run sttr moet deno aanroepen
+// XXX document...
+if (Deno.args.length < 1) {
+  console.error(`
+Dev usage:
+$ deno run --unstable --allow-net --allow-write --allow-read transform.ts BASE_DIR --config=config.local.ts (default=config.ts)
+`);
+  Deno.exit();
+}
+
+const baseDir = join(Deno.cwd(), Deno.args[0]);
+const { apis }: { apis: APIConfig[] } = await import(
+  join(Deno.cwd(), "config.local.ts")
+); // XXX convert to cli argument
+
+const permitIds: string[] = apis.flatMap((api: APIConfig) =>
+  Object.values(api.topics)
+);
+
+const apisMap: object[] = apis.map(async (api: APIConfig) => {
+  const { outputDir, version } = api;
+  const topics: { [key: string]: string[] } = api.topics;
+
+  const apiPermits: any = await readJson(join(baseDir, outputDir, "list.json"));
+  const apiPermitIds: string[] = apiPermits.map(({ _id }: any) => _id);
+
+  apiPermitIds.forEach((permitId) => {
+    if (permitIds.indexOf(permitId) === -1) {
+      topics[permitId] = [permitId];
+    }
+  });
+
+  // XXX duplicate parsing of xml's for topics defined in config
+  // XXX i want them in separate sttr-json's but only parse them once?
+
+  // Put json-sttr-version from flolegal in our sttr-json
+
+  /* Convert multiple permit-xml files per topic to a json string and write to the [topic].json file */
+  Object.entries(topics).forEach(([slug, permits]: [string, string[]]) => {
+    const topicJsonContent = {
+      slug,
+      permits: permits.map(async (permitId) => {
+        const apiPermit = apiPermits.find(
+          (apiPermit: any) => apiPermit._id === permitId
```

Comment:
any?

Reply:
Yeah, I could use some help here. I'll address it in the next Dev-meeting.